observability/akka-http-metrics/src/test/suite/scala/com/daml/metrics/akkahttp/AkkaHttpMetricsSpec.scala

```diff
@@ -0,0 +1,621 @@
+      }
+    }
+
+    "record byte payload size on failed (missing route) response" in {
+      withRouteAndMetrics { (route, metrics) =>
+        Get(
+          "/undefined",
+          HttpEntity.Strict(ContentTypes.`application/octet-stream`, ByteString(byteArray1)),
+        ) ~> route ~> check {
+          val response = responseAs[String]
+          metrics.httpResponsesBytesTotalValue should be(
+            response.length.toLong
+          )
+        }
+      }
+    }
+
+    "record byte payload size on failed (unauthorized) response" in {
+      withRouteAndMetrics { (route, metrics) =>
+        Get(
+          "/unauthorized",
+          HttpEntity.Strict(ContentTypes.`application/octet-stream`, ByteString(byteArray1)),
+        ) ~> route ~> check {
+          val response = responseAs[String]
+          metrics.httpResponsesBytesTotalValue should be(
+            response.length.toLong
+          )
+        }
+      }
+    }
+
+    "record byte payload size on failed (bad request) response" in {
+      withRouteAndMetrics { (route, metrics) =>
+        Get(
+          "/badrequest",
+          HttpEntity.Strict(ContentTypes.`application/octet-stream`, ByteString(byteArray1)),
+        ) ~> route ~> check {
+          val response = responseAs[String]
+          metrics.httpResponsesBytesTotalValue should be(
+            response.length.toLong
+          )
+        }
+      }
+    }
+
+    "record byte payload size on failed (exception) response" in {
+      withRouteAndMetrics { (route, metrics) =>
+        Get(
+          "/exception",
+          HttpEntity.Strict(ContentTypes.`application/octet-stream`, ByteString(byteArray1)),
+        ) ~> route ~> check {
+          val response = responseAs[String]
+          metrics.httpResponsesBytesTotalValue should be(
+            response.length.toLong
+          )
+        }
+      }
+    }
+
+    "record byte payload size on multiple response" in {
+      withRouteAndMetrics { (route, metrics) =>
+        Get(
+          "/",
+          HttpEntity.Strict(ContentTypes.`text/plain(UTF-8)`, ByteString("hello")),
+        ) ~> route ~> check {
+          responseAs[String] // force processing the response
+        }
+        Get(
+          "/mirror/200",
+          HttpEntity.Chunked(
+            ContentTypes.`application/octet-stream`,
+            Source(List(ByteString(byteArray1), ByteString(byteArray2))),
+          ),
+        ) ~> route ~> check {
+          responseAs[String] // force processing the response
+        }
+        Get(
+          "/exception",
+          HttpEntity.Strict(ContentTypes.`application/octet-stream`, ByteString(byteArray1)),
+        ) ~> route ~> check {
+          val response = responseAs[String]
+          metrics.httpResponsesBytesTotalValue should be(
+            4L + 15L + response.length.toLong
+          )
+        }
+      }
+    }
+  }
+
+  "request_bytes_total and responses_bytes_total support" should {
+    "not alter string data" in {
```

Comment:
Aren't we testing here the test code, i.e., whether the `/mirror` path implementation duplicates input properly?